src/components/oculus-touch-controls.js

```diff
@@ -270,48 +274,81 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
-
-    // Update trigger and/or grip meshes, if any.
-    if (buttonMeshes) {
-      if (button === 'trigger' && buttonMeshes.trigger) {
-        buttonMeshes.trigger.rotation.x = this.originalXRotationTrigger - analogValue * (Math.PI / 26);
-      }
-      if (button === 'grip' && buttonMeshes.grip) {
-        buttonMeshes.grip.position.x = this.originalXPositionGrip + (this.data.hand === 'left' ? -1 : 1) * analogValue * 0.004;
+    // move the button meshes
+    if (this.isV3) {
+      this.onButtonChangedV3(evt);
+    } else {
+      var button = this.mapping[this.data.hand].buttons[evt.detail.id];
+      var buttonMeshes = this.buttonMeshes;
+      var analogValue;
+      if (!button) { return; }
+
+      if (button === 'trigger' || button === 'grip') { analogValue = evt.detail.state.value; }
+
+      if (buttonMeshes) {
+        if (button === 'trigger' && buttonMeshes.trigger) {
+          buttonMeshes.trigger.rotation.x = this.originalXRotationTrigger - analogValue * (Math.PI / 26);
+        }
+        if (button === 'grip' && buttonMeshes.grip) {
+          analogValue *= this.data.hand === 'left' ? -1 : 1;
+          buttonMeshes.grip.position.x = this.originalXPositionGrip + analogValue * 0.004;
+        }
       }
     }
-
     // Pass along changed event with button state, using the buttom mapping for convenience.
     this.el.emit(button + 'changed', evt.detail.state);
   },
 
-  onModelLoaded: function (evt) {
-    var controllerObject3D = this.controllerObject3D = evt.detail.model;
-    var buttonMeshes;
+  clickButtons: ['xbutton', 'ybutton', 'abutton', 'bbutton', 'thumbstick'],
+  onButtonChangedV3: function (evt) {
+    var button = this.mapping[this.data.hand].buttons[evt.detail.id];
+    var buttonObjects = this.buttonObjects;
+    var analogValue;
+    if (!button) { return; }
 
-    if (!this.data.model) { return; }
+    analogValue = evt.detail.state.value;
+    analogValue *= this.data.hand === 'left' ? -1 : 1;
+
+    if (button === 'trigger') {
+      this.triggerEuler.copy(this.buttonRanges.trigger.min.rotation);
+      this.triggerEuler.x += analogValue * this.buttonRanges.trigger.diff.x;
+      this.triggerEuler.y += analogValue * this.buttonRanges.trigger.diff.y;
+      this.triggerEuler.z += analogValue * this.buttonRanges.trigger.diff.z;
+      buttonObjects.trigger.setRotationFromEuler(this.triggerEuler);
+    } else if (button === 'grip') {
+      buttonObjects.grip.position.x = buttonObjects.grip.minX + analogValue * 0.004;
+    } else if (this.clickButtons.includes(button)) {
+      buttonObjects[button].position.y = analogValue === 0 ? this.buttonRanges[button].unpressedY : this.buttonRanges[button].pressedY;
+    }
+  },
 
-    buttonMeshes = this.buttonMeshes = {};
+  onModelLoaded: function (evt) {
+    if (this.isV3) {
+      this.onModelLoadedV3(evt);
+    } else {
+      var controllerObject3D = this.controllerObject3D = evt.detail.model;
```

Comment:
Probably can factor out this part to a separate function. onOculusTouchV1ModelLoaded?

Reply:
Well, to be clear, this handles three versions of oculus headsets that came before "v3". Seems if you call it v1 is might be confusing?

Reply:
I see. Maybe a comment in the branch will help clarify

Reply:
Do you mean just a line between 328 and 329 like
`// All oculus headset controller models prior to the Quest 2 (i.e., Oculus Touch V3) used a consistent format that is handled here` ?

Or something else / somewhere else?

Reply:
yep